### PR TITLE
Finalize listings definition

### PR DESCRIPTION
**Pull request by andr-dev**
Updating listings request definition to include location, bedrooms total / available, bathrooms total / available etc.

### CODE REVIEW

File: android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageViewModel.kt [hunk 1]
```diff
@@ -105,19 +109,28 @@ class CreateListingPageViewModel @Inject constructor(
...
 							addressPostalcode = it.address.addressPostalCode,
 							addressCountry = it.address.addressCountry,
 							price = it.price,
-							rooms = it.numBedrooms,
 							leaseStart = it.startDate,
 							leaseEnd = it.endDate,
 							description = it.description,
 							residenceType = ResidenceType.house,
 							imgIds = imgIds,
+							latitude = location!!.latitude.toFloat(),
```
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageViewModel.kt]: Can we maybe not take down the entire app if there is no location?

File: android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageViewModel.kt [hunk 2]
```diff
@@ -23,19 +24,28 @@ import javax.inject.Inject
...
+		BehaviorSubject.createDefault(false)
 	private val listingDetailsStream: Observable<Result<ListingDetails>> = listingIdStream.map {
 		runCatching {
 			runBlocking {
-				val listing = listingsApi.listingsDetails(it)
+				val location = locationService.getLocation()
+				val listing =
+					listingsApi.listingsDetails(
+						listingId = it,
+						longitude = location!!.longitude.toFloat(),
```
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageViewModel.kt]: Can we maybe not take down the entire app if there is no location?

**Review by leeminkyu16 — APPROVED** [review]
Let's try not to destroy the entire app if location happens to be null.
Otherwise, Let's Gamble, Try Merging

File: android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageViewModel.kt (hunk 2 above)
Comment (Jiay0928) [reply android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageViewModel.kt]: Lol, I can use the location of like  university of waterloo as default

**Comment by leeminkyu16** [comment]
I'll rebase since I was a big meanie and I merged my changes in before you

**Review by henryhu1 — APPROVED** [review]
I've been working on getting latitude and longitude of the sublet location, pretty sure I can provide it to the apis once I sync changes

**Comment by leeminkyu16** [comment]
I hope I didn't mess up the rebase lol

File: android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageViewModel.kt [hunk 3]
```diff
@@ -105,19 +109,28 @@ class CreateListingPageViewModel @Inject constructor(
...
+							longitude = location.longitude.toFloat(),
```
Comment (henryhu1) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageViewModel.kt]: why does this line not have `!!`?

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/HomeListChildViewModel.kt [hunk 4]
```diff
@@ -79,15 +81,33 @@ class HomeListChildViewModel @Inject constructor(
...
 					runBlocking {
+						val location = locationService.getLocation()
 						// TODO: Change to use filter values
 						listingsApi.listingsList(
-							priceMin = getListingParams.priceRange.lowerBound,
-							priceMax = getListingParams.priceRange.upperBound,
-							roomsMin = null,
-							roomsMax = null,
+							longitude = location!!.longitude.toFloat(),
+							latitude = location.latitude.toFloat(),
```
Comment (henryhu1) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/HomeListChildViewModel.kt]: here too `!!`

File: android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageViewModel.kt (hunk 3 above)
Comment (andr-dev) [reply android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageViewModel.kt]: I think if the previously executed location!! succeeds, then kotlin can tell that location is non-null and will compile.

**Review by andr-dev — COMMENTED** [review]
Fixed the rebase

File: android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageViewModel.kt (hunk 3 above)
Comment (leeminkyu16) [reply android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageViewModel.kt]: Yeah, Kotlin has "smart-casting" which allows Kotlin to deduce that if line 125 doesn't destroy the app, then location must not be null

Comment (henryhu1) [reply android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageViewModel.kt]: I see 
if it runs then no issue

**Review by henryhu1 — APPROVED** [review]
looks good to me
getting ready to grind it out

File: android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageViewModel.kt (hunk 2 above)
Comment (leeminkyu16) [reply android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageViewModel.kt]: Yeah, or you can just not make the call if the location is missing

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/HomeListChildViewModel.kt [hunk 5]
```diff
@@ -80,14 +82,31 @@ class HomeListChildViewModel @Inject constructor(
...
 					runBlocking {
+						Log.d("lowerBound",getListingParams.locationRange.lowerBound?.toFloat().toString())
 						// TODO: Change to use filter values
 						listingsApi.listingsList(
-							priceMin = getListingParams.priceRange.lowerBound,
-							priceMax = getListingParams.priceRange.upperBound,
-							roomsMin = null,
-							roomsMax = null,
+//						TODO: ADDING LOCATION VALUE
+							longitude = 0f,
```
Comment (Jiay0928) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/HomeListChildViewModel.kt:89]: LocationService is acting wired and crashing the app, so currently hardcoded the value, will update the location later in a separate PR.

File: android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageViewModel.kt (hunk 1 above)
Comment (andr-dev) [reply android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageViewModel.kt]: Moving to separate PR.

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/HomeListChildViewModel.kt (hunk 5 above)
Comment (leeminkyu16) [reply android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/HomeListChildViewModel.kt:89]: Ah, yeah, `LocationService` does that lol... 😭 
If you're talking about it crashing the entire app despite putting it in a different thread with a try catch around it that is.

**Review by leeminkyu16 — APPROVED** [review]
Let's Gamble, Try Merging

**Pull request merged**